graphql-server/src/dataStore/dataStore.service.ts

```diff
@@ -69,7 +69,9 @@ export class DataStoreService {
     const existing = await repo.findOneBy({ name: item.name });
 
     if (existing) {
-      if (existing.connectionUrl === item.connectionUrl) return undefined;
+      // Handles connection URLs in config that contain database name
+      if (item.connectionUrl.startsWith(existing.connectionUrl))
```

Comment:
This might need to be more specific. What if I'm trying to start a connection with the same host but different ports? You'd hit this exception instead of getting an error I think